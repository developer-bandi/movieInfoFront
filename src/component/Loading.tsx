import styled from 'styled-components';
import loading from '../tempdata/loading.gif';
const Loading = () => {
  return (
    <LoadingBlock>
      <Loadingimg src={loading} />;
    </LoadingBlock>
  );
};

const LoadingBlock = styled.div`
  background: #fafafa;
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
`;
const Loadingimg = styled.img`
  margin: auto;
  margin-top: 6rem;
  margin-bottom: 6rem;
`;

export default Loading;
