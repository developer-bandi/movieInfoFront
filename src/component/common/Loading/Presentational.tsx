import loading from '../../../imgs/loading.gif';
import styles from './Style';
const LoadingPresentational = () => {
  return (
    <styles.MainBlock>
      <styles.Img src={loading} data-testid={'loading'} />;
    </styles.MainBlock>
  );
};

export default LoadingPresentational;
