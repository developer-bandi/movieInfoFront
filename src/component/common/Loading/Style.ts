import styled from 'styled-components';

const MainBlock = styled.div`
  background: #fafafa;
  width: 100%;
  height: 1000px;
  display: flex;
  justify-content: center;
`;
const Img = styled.img`
  width: 500px;
  height: 500px;
  margin: auto;
  margin-top: 6rem;
  margin-bottom: 6rem;
`;

const styles = { MainBlock, Img };

export default styles;
