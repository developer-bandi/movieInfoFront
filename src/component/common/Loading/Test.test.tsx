import { render } from '@testing-library/react';
import Loading from './Presentational';

describe('<Loading/> Presentational', () => {
  it('전체 화면 검증', () => {
    const utils = render(<Loading />);
    expect(utils.container).toMatchSnapshot();
  });
});
