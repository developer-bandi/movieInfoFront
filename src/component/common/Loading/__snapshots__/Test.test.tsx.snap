// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading/> Presentational Loading 스냅샷 비교 1`] = `
<div>
  <div
    class="sc-bdvvtL ieFVTS"
  >
    <img
      class="sc-gsDKAQ bgYcDx"
      data-testid="loading"
      src="loading.gif"
    />
    ;
  </div>
</div>
`;

exports[`<Loading/> Presentational 전체 화면 검증 1`] = `
<div>
  <div
    class="sc-bdvvtL ieFVTS"
  >
    <img
      class="sc-gsDKAQ bgYcDx"
      data-testid="loading"
      src="loading.gif"
    />
    ;
  </div>
</div>
`;
